cmd/update.go

```diff
@@ -90,12 +90,24 @@ var updateCmd = &cobra.Command{
 			// we want to use and save absolute path for component
 			dir, err := filepath.Abs(componentLocal)
 			checkError(err, "")
+			fileInfo, err := os.Stat(dir)
+			checkError(err, "")
+			if !fileInfo.IsDir() {
+				fmt.Println("Please provide path to a directory")
+				os.Exit(1)
+			}
 			err = component.Update(client, componentName, "local", dir)
 			checkError(err, "")
 			fmt.Printf("The component %s was updated successfully\n", componentName)
 		} else if len(componentBinary) != 0 {
 			path, err := filepath.Abs(componentBinary)
 			checkError(err, "")
+			fileInfo, err := os.Stat(path)
```

Comment:
It should be possible to create binary component even if binary file doesn't exist #436

Reply:
What about create.go then?

Reply:
for create.go the same. I actually wanted to post my comment for create.go.

for the local component, we can verify that the directory exists.  But for binary component, it is ok, if the binary file doesn't exist at the time when `odo create` or `odo update` is run. that check should be done only for `odo push`

Reply:
OK :+1:

Reply:
@kadel What about the isDir thing? Should that be removed too?